lineapy/plugins/taskgen.py

```diff
@@ -43,9 +45,9 @@ def get_task_definitions(
         )
 
 
-def get_artifact_task_definitions(
+def get_artifact_task_definition_graph(
     artifact_collection: ArtifactCollection, pipeline_name: str
-) -> Dict[str, TaskDefinition]:
+) -> Tuple[Dict[str, TaskDefinition], TaskGraph]:
```

Comment:
We can simply this by returning "augmented" `TaskGraph` where each task node has the corresponding `TaskDefinition` "attached" as an attribute. Specifically, I am thinking we can do:

```python
def get_artifact_task_definition_graph(
    artifact_collection: ArtifactCollection, pipeline_name: str
) -> TaskGraph:

    [...]

    # "Attach" each task definition to the corresponding node in task graph
    for taskname in task_graph.nodes:
        task_graph.nodes[taskname]["task_definition"] = task_definitions[taskname]

    return task_graph
```

And the same for `get_session_task_definition_graph()` and `get_allsessions_task_definition_graph()` as well.

Reply:
I couldn't actually do this, the nodes in TaskGraph need to be hashable and TaskDefinitions are not.

Reply:
I see, that's too bad! Then, let's proceed with the current solution; we may find a better alternative later.